src/main/scala/is/hail/utils/package.scala

```diff
@@ -473,4 +473,27 @@ package object utils extends Logging
     }
     expansion
   }
+
+  def compress(input: Array[Byte]): Array[Byte] = {
+    val compressor = new Deflater()
+    compressor.setInput(input)
+    compressor.finish()
+
+    var compressedData = new Array[Byte](1024)
+    val buffer = new Array[Byte](1024)
+    var compressedOffset = 0
+
+    while (!compressor.finished()) {
+      val nBytesCompressed = compressor.deflate(buffer)
+
+      while (nBytesCompressed > compressedData.length - compressedOffset) {
+        compressedData = java.util.Arrays.copyOf(compressedData, compressedData.length << 1)
```

Comment:
nBytesCompressed can't be bigger than the length of buffer (1024), so you only need to do at most one step of this loop.